### PR TITLE
Tutor registration

### PR DESCRIPTION
**Pull request by tjinks89**
Tutors can now register as well

### CODE REVIEW

**Comment by runocan** [comment]
I checked this out and I think the conflict might be moving auth.dart to the backend folder but I'm not sure. I don't want to merge this just yet because there lots of new stuff added. Could someone else check as well? Please and thank you.

File: lib/backend/auth.dart [hunk 1]
```diff
@@ -45,12 +45,25 @@ class Auth {
...
+
+  Future registerTutor(String email, String password, String firstName, String lastName, String major) async {
+    try {
+      AuthResult result = await _auth.createUserWithEmailAndPassword(email: email, password: password);
+      FirebaseUser user = result.user;
+      // create a new Tutor for the user with the uid
+      await DatabaseService(uid: user.uid).updateTutorData(email, firstName, lastName, major);
       return _userFromFirebaseUser(user);
     } catch (error) {
       print(error.toString());
```
Comment (coolfortune) [inline lib/backend/auth.dart]: Duplicate code 1
lines 48 to 70

File: lib/backend/database.dart [hunk 2]
```diff
@@ -18,16 +20,25 @@ class DatabaseService {
...
-        email: doc.data['email'] ?? ''
-      );
-    }).toList();
+  Future<void> updateTutorData(String email, String firstName, String lastName, String major) async {
+    return await tutorCollection.document(uid).setData({
+      'email': email,
+      'firstName': firstName,
+      'lastName' : lastName,
+      'major': major,
+    });
```
Comment (coolfortune) [inline lib/backend/database.dart]: Duplicate code 2

File: lib/signup.dart [hunk 3]
```diff
@@ -21,17 +21,39 @@ class _SignupPage extends State<SignupPage> {
...
+      else if (_type == StudentType.tutor)
+      {
+        print('registering tutor');
+        dynamic result = await _auth.registerTutor(_email.trim(), _password, _firstName, _lastName, _major);
+        if (result == null)
+        {
+          print('registration failed');
+        }
+      }
+    }
```
Comment (coolfortune) [inline lib/signup.dart:52]: Duplicate code 3
Also consider adding an alert then the user registers successfully or unsuccessfully (I used a SnackBar when I did an alert for my Forgot Password page, but you might have to wrap it in a Builder widget)

**Comment by aguilarjavier** [comment]
We are now making users and tutors! I only have 2 things to pick on. One is smaller and its just that it would be nice if once you click on sign up it takes you to the log in page or even automatically logs you in. Instead of keeping you on the sign up page.
The other thing is the new user themselves in the collection. From when I tried it out, when we click sign up as a tutor, for example, the fields for ratings and schedule and all that arent in the collection. Only the fields we filled out in the form are there. If we could add those missing fields as empty from the beginning and have them be already in the database then that would make those account ready to use right as you sign up. 
As far as the conflicts go. From my first look they seem to just be from things being moved to a backend file. That should be an easy fix and VS does a good job at automatically fixing imports. The conflicts in logIn.dart are also just from imports.

**Review by coolfortune — APPROVED** [review]
Good Job, just have a few comments listed, mostly the code duplication, but since were short on time, don't rush to make these changes.

Also, be careful handing the merge conflicts

**Review by aguilarjavier — APPROVED** [review]

**Comment by tjinks89** [comment]
Javier, I absolutely agree with both ideas...at the time I didn't even have those extra fields for the students or tutors and I think that can be an easy fix on my end. I have not yet done anything with transitioning to new pages on state changes...you want to run with that while I fix the collections being updated?

**Review by runocan — APPROVED** [review]

**Pull request merged**